hack/netpol/README.md

```diff
@@ -0,0 +1,136 @@
+z/a	X	.	.	.	.	.	.	.	.
+x/b	.	.	.	.	.	.	.	.	.
+y/b	.	.	.	.	.	.	.	.	.
+z/b	.	.	.	.	.	.	.	.	.
+x/c	X	.	.	.	.	.	.	.	.
+y/c	X	.	.	.	.	.	.	.	.
+z/c	X	.	.	.	.	.	.	.	.
+
+
+observed:
+
+-	x/a	y/a	z/a	x/b	y/b	z/b	x/c	y/c	z/c
+x/a	.	.	.	.	.	.	.	.	.
+y/a	X	.	.	.	.	.	.	.	.
+z/a	X	.	.	.	.	.	.	.	.
+x/b	.	.	.	.	.	.	.	.	.
+y/b	.	.	.	.	.	.	.	.	.
+z/b	.	.	.	.	.	.	.	.	.
+x/c	X	.	.	.	.	.	.	.	.
+y/c	X	.	.	.	.	.	.	.	.
+z/c	X	.	.	.	.	.	.	.	.
+
+
+comparison:
+
+-	x/a	y/a	z/a	x/b	y/b	z/b	x/c	y/c	z/c
+x/a	.	.	.	.	.	.	.	.	.
+y/a	.	.	.	.	.	.	.	.	.
+z/a	.	.	.	.	.	.	.	.	.
+x/b	.	.	.	.	.	.	.	.	.
+y/b	.	.	.	.	.	.	.	.	.
+z/b	.	.	.	.	.	.	.	.	.
+x/c	.	.	.	.	.	.	.	.	.
+y/c	.	.	.	.	.	.	.	.	.
+z/c	.	.	.	.	.	.	.	.	.
+
+```
+
+## How is this different then NetworkPolicy tests in upstream K8s ?
+
+We are working to merge this into upstream Kubernetes, in the meanwhile, here's the differences.
+
+- We define tests as *truth tables, and have a 'builder' library* for building up network policy structs with almost no boilerplate, meaning you can define a very sophisticated network policy test in just a few lines of code.
+- *Comprehensive:* All pod-to-pod connectivity is validated for every test run.  In a typical network policy test in current upstream we only validate 2 or 3 scenarios, leaving out intra and inner namespace connections which might be compromised due to a hard to detect CNI inconsistency.  In these tests, we test all 81 connections for 3 identical pods running in 3 different namespaces (i.e. the 9x9 connectivity matrix).
+- *Transparent:* Each test prints out a `kubectl` command you can run to re-probe a given pods connectivity patterns.
+- It's *fast:* Because we use `kubectl exec` to run tests with `wget` between pods, all 81 tests can easily finish in 20 seconds or less, even if pod scheduling is slow.  This is because no polling is done, and there is no down/uptime for pods.
+- *Easy to reason about:* The pods in this repo stay up forever, so you can reuse the above kubectl commands outputted by your netpol logs to exec into a pod and reproduce any failures.
+- *Scalable:* If you want to test 32 policies, all at once ? Just take a look at the example test (in `main`) and copy paste a few lines, and you'll be testing enterprise CNI application patterns in a heartbeat.
+
+## Users
+
+### Test Your Dang CNI !  Now !
+
+Create the policy probe tests:
+
+```
+kubectl create clusterrolebinding netpol --clusterrole=admin --serviceaccount=kube-system:netpol
+kubectl create sa netpol -n kube-system
+kubectl create -f https://raw.githubusercontent.com/vmware-tanzu/antrea/master/hack/netpol/install.yml
+```
+
+Now, look at the results of the network policy probe:
+
+```
+ kubectl logs `kubectl get pods -n kube-system | grep netpol | cut -d' ' -f 1` -n kube-system  
+```
+ 
+## Developers
+
+This is a new library for building complex, comprehensive network policy tests.  To build it, cd to antrea/ and run `go build -o hack/netpol/pkg/main/main.go`.  Then, execute the binary.
```

Comment:
The `go build -o hack/netpol/pkg/main/main.go` doesn't look right to me. IIRC correctly, `-o` requires an output directory. Maybe just suggest devs to use `go run hack/netpol/pkg/main/main.go` directly.

Reply:
done